pep-0387.txt

```diff
@@ -112,6 +112,28 @@ Basic policy for backwards compatibility
   platforms).
 
 
+Soft Deprecation
+================
+
+A soft deprecation can be used when using an API must no longer be used
+to write new code, but it remains safe to continue using it in existing
+code. The API remains documented and tested, but will not be developed
+further (no enhancement).
+
+The documentation of a soft deprecation should explain why the API
+should be avoided, and if possible propose a replacement.
+
+Soft deprecated functions must emit a SoftDeprecationWarning warning
```

Comment:
What about soft deprecated *modules* (like optparse, for example)? It would seem better to issue this warning on import, and not have every individual function call issue a warning.

Reply:
I clarified this point.